### ISSUE TITLE
[feature request] trigger webhooks

### ISSUE THREAD

**Issue by 4383** [hoarder-app/hoarder#384]
Hey there,

I'm looking for a bookmark solution that allow me to trigger webhooks to a specified url once I bookmark a thing (a note, an article, a blog post et....).

AFAIK this is not feasible with hoarder.
Do you think it would be possible to implement this kind of feature?

Thanks for all the work you have already done.

**Comment by MohamedBassem** [comment]
It's indeed currently not possible. Do you mind expanding a bit on how you'd use this?

**Comment by 4383** [comment]
Sure.

Here is my personal use case:

I daily read books, during my reading I take notes and store links into hoarder. 
I store my personal notes by using the [obsidian markdown editor](https://obsidian.md/) to create my own [zettlekasten](https://en.wikipedia.org/wiki/Zettelkasten) (an obsidian vault).

But today to automatize my note taking process, I created my own system:
- I launched a local instance of the llama 3.1 LLM.
- I connected my llama to a telegram bot, and, when during my reading, I found a useful concept, I send the concept name as a message to my telegram bot, and then, this bot trigger a llama prompt
- once the LLM prompt is done, the bot create a new note in my obisidian vault. A note related to the concept I passed as message and built with the output of llama.

My knowledge base is growing automatically in parallel of my reading and I have no friction due to note taking during learning.

If I'm able to simply store in hoarder the link of web page explaining the concept, by example a wikipedia page, and then trigger web hook to my local LLM endpoint, then I'd be able to remove lot of mechanisms I used to create this workflow.

The ideal workflow would be:
1. I create a bookmark;
2. the LLM prompt is triggered by the hoarder webhook on bookmarks additions (by passing the title of bookmark);
3. my note is automatically created in my vault on the result of the LLM prompt request.

My LLM endpoint would be exposed by a script to manage the IO operations (the local file creation).

Let me know if you need more details.

**Comment by ebenoist** [comment]
I would love to see this or just an RSS feed of my hoarded items. I had been building a similar application to hoarder before I discovered it and quickly abandoned my plans of building my own (thank you). Here's the code I used to set up a simple RSS feed based on the items I've saved.

If you're interested I could probably port this functionality to hoarder itself.

https://github.com/ebenoist/enlace

**Comment by MohamedBassem** [comment]
@ebenoist Producing an RSS feed is tracked in #333.  To be honest, it's currently on not my high priority list, but I'd be open to accepting a contribution if you're up for it :) You can join us on the #development channel in discord to discuss the specifics if you're interested.

**Comment by ebenoist** [comment]
@MohamedBassem I'm down to help out. I don't think it'd be difficult. I have the basics, I just need to rewrite it in typescript. I'll jump in the channel if I have any questions!

**Comment by MohamedBassem** [comment]
This got implemented in #852 and will be available in the next release.

**Comment by 4383** [comment]
Thanks! Much appreciated
